### DISCUSSION TITLE
RBAC improvement by processing roles as an alias for it's users

### DISCUSSION THREAD

**Discussion by TechAuditBI** [apache/superset#25793] [Ideas]
The idea is very simple: make it possible to select roles in every field were users can be selected in Superset. Including fields like "owner" in charts parameters and "users" in roles parameters. Selecting a role in those fields should be equal to selecting all users with this role.
This feature would: 
1. make it much easier to grant access to large amounts of users
2. make it much easier to mantain access to content via users groups up to date
3. allow roles to be nested

If more detailed description is needed please comment.

I think this feature would practically cover all the points in requests like this [#18556](https://github.com/apache/superset/discussions/18556)

**Comment by rusackas** [comment]
I think @villebro is going to be making a proposal along these lines, as discussed in SIP Office Hours.

**Reply by villebro** [reply]
Thanks for pinging me @rusackas - @TechAuditBI I am indeed working on solving precisely this problem, and I'm going to be opening a SIP along with a prototype PR that adds this capability. I'll ping you when it's ready for review/comments.
